apps/backend/src/services/audit_service.py

```
@@ -0,0 +1,443 @@
+            if not dry_run and count_to_delete > 0:
+                # Delete old entries
+                from sqlalchemy import delete
+                delete_query = delete(DataCollectionAudit).where(DataCollectionAudit.time < cutoff_date)
+                await self.db.execute(delete_query)
+                await self.db.commit()
+
+            return count_to_delete
+
+        except Exception as e:
+            logger.error(f"Error cleaning up audit entries: {e}")
+            raise DatabaseOperationError(f"Failed to cleanup audit entries: {str(e)}", "cleanup_old_entries")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)